### ISSUE TITLE
Mixed-content warnings on /search

### ISSUE THREAD

**Issue by robertknight** [hypothesis/h#4084]
The /search pages frequently display Mixed Content warnings in the browser - usually due to the presence of non-HTTPS image links in annotations:

![mixed-content-warning](https://cloud.githubusercontent.com/assets/2458/20346362/73ea484a-abf3-11e6-91d9-a082cf849742.png)

Given the increased prevalence of HTTPS support on the web, I propose that we automatically convert all HTTP links in media on annotations to HTTPS. We could optionally add some fallback if the image fails to load.

**Comment by robertknight** [comment]
You can argue that this is working as intended and thus not a bug, but the end result is a fairly prominent warning from the browser telling the user that the page is not secure.

**Comment by nickstenning** [comment]
Yyyep. We need to deploy something like [camo](https://github.com/atmos/camo).

**Comment by robertknight** [comment]
To be clear - my hypothesis is that the majority of image or other media in annotations that have an HTTP scheme are also available over HTTPS. If that is true, then perhaps we could just rewrite the URL and avoid running extra infrastructure?

**Comment by nickstenning** [comment]
> my hypothesis is that the majority of image or other media in annotations that have an HTTP scheme are also available over HTTPS

Even if that's true for 90% of them, it's doesn't seem ok to me to break 10% of the image links people post.

**Comment by robertknight** [comment]
> Even if that's true for 90% of them, it's doesn't seem ok to me to break 10% of the image links people post.

My original thinking was that if the percentage was very small (say 2-5%) we could have a fallback that would just display a link to open the image in a new tab if loading the image failed.

However, it looks like the percentage that can be rewritten is much lower.

I did an investigation where I scraped the URLs from the bodies of 50K recent public annotations, filtered those down to HTTP URLs and attempted to fetch both the original URL and the URL with scheme rewritten to HTTPS. From a sample of 1307 URLs for which the original HTTP URL returned a 200 OK, only 643 (~49%) returned a 200 OK (note, I'm not even checking for an actual content match here) when fetching the same URL except with an HTTPS scheme.
